Sources/Database/Database.swift

```diff
@@ -391,6 +395,23 @@ private struct _Client {
       .map(const(unit))
   }
 
+  func updateEpisodeProgress(
+    episodeSequence: Episode.Sequence,
+    percent: Int,
+    userId: Models.User.Id
+  ) -> EitherIO<Error, Prelude.Unit> {
+    return self.execute(
+      #"""
+      INSERT INTO "episode_progresses" ("episode_sequence", "percent", "user_id")
+      VALUES ($1, $2, $3)
+      ON CONFLICT ("episode_sequence", "user_id") DO UPDATE
+      SET "percent" = GREATEST(episode_progresses.percent, $2)
```

Comment:
this makes sure that if a lower percentage comes in we won't count it towards their progress.